src/misc_gui.cpp

```diff
@@ -97,21 +92,19 @@ class LandInfoWindow : public Window {
 		if (widget != WID_LI_BACKGROUND) return;
 
 		size->height = WD_TEXTPANEL_TOP + WD_TEXTPANEL_BOTTOM;
-		for (uint i = 0; i < LAND_INFO_CENTERED_LINES; i++) {
-			if (StrEmpty(this->landinfo_data[i])) break;
-
-			uint width = GetStringBoundingBox(this->landinfo_data[i]).width + WD_FRAMETEXT_LEFT + WD_FRAMETEXT_RIGHT;
-			size->width = std::max(size->width, width);
-
-			size->height += FONT_HEIGHT_NORMAL + WD_PAR_VSEP_NORMAL;
-			if (i == 0) size->height += 4;
-		}
-
-		if (!StrEmpty(this->landinfo_data[LAND_INFO_MULTICENTER_LINE])) {
-			uint width = GetStringBoundingBox(this->landinfo_data[LAND_INFO_MULTICENTER_LINE]).width + WD_FRAMETEXT_LEFT + WD_FRAMETEXT_RIGHT;
-			size->width = std::max(size->width, std::min(300u, width));
-			SetDParamStr(0, this->landinfo_data[LAND_INFO_MULTICENTER_LINE]);
-			size->height += GetStringHeight(STR_JUST_RAW_STRING, size->width - WD_FRAMETEXT_LEFT - WD_FRAMETEXT_RIGHT);
+		for (size_t i = 0; i < this->landinfo_data.size(); i++) {
+			if (i == this->landinfo_data.size() - 1 && this->last_is_multicenter) {
```

Comment:
What is the reason you have chosen to introduce `last_is_multicenter` and some fudging with the element being the last in the vector instead of having a `cargo_acceptance` string where you put `line.str()` in and check the string for emptiness like was done before?

Reply:
Because I didn't think of it. It seems a lot nicer, though. Expect an update soon.